### PR TITLE
add audit logs API

### PR DESCRIPTION
**Pull request by wy65701436**
1, add API entry for get audit logs
2. add audit log manager to hanlder CRUD

Use the new format of audit log to cover differernt resource, artifact/tag/repostory/project

Signed-off-by: wang yan <wangyan@vmware.com>

### CODE REVIEW

File: make/migrations/postgresql/0030_2.0.0_schema.up.sql [hunk 1]
```diff
@@ -173,3 +173,14 @@ CREATE TABLE artifact_2
...
   CONSTRAINT unique_artifact_2 UNIQUE (project_id, repo, tag)
 );
+
+CREATE TABLE audit_log
+(
+ id             SERIAL PRIMARY KEY NOT NULL,
+ project_id     int NOT NULL,
+ operation      varchar(20) NOT NULL,
+ resource_type  varchar(255) NOT NULL,
+ resource       varchar(255) NOT NULL,
```
Comment (ywk253100) [inline make/migrations/postgresql/0030_2.0.0_schema.up.sql]: The length of repository+digest may exceeds  255

File: api/v2.0/swagger.yaml [hunk 2]
```diff
@@ -499,6 +499,68 @@ paths:
           $ref: '#/responses/409'
         '500':
           $ref: '#/responses/500'
+  /auditlogs:
```
Comment (ywk253100) [inline api/v2.0/swagger.yaml]: According to our internal REST API standard `URIs SHOULD use kebab-case`, `/audit-logs` is better

File: make/migrations/postgresql/0030_2.0.0_schema.up.sql (hunk 1 above)
Comment (wy65701436) [reply make/migrations/postgresql/0030_2.0.0_schema.up.sql]: extend it to 1024

File: api/v2.0/swagger.yaml (hunk 2 above)
Comment (wy65701436) [reply api/v2.0/swagger.yaml]: updated, thx.

**Pull request merged**